src/options/attribute-sorting/utils.ts

```diff
@@ -22,47 +22,38 @@ function compareByIndex(leftIndex: number, rightIndex: number): CompareResult {
 export function compareAttributeToken(
 	a: AttributeToken,
 	b: AttributeToken,
-	sortAttributes: string[],
-	moveToEnd: boolean = false
+	sortAttributesEnd: string[],
+	sortAttributesBeginning: string[]
 ): CompareResult {
-	const sortPatterns: RegExp[] = sortAttributes.map((sort) => new RegExp(sort));
+	const sortPatternsBeginning: RegExp[] = sortAttributesBeginning.map((sort) => new RegExp(sort)).reverse();
+	const sortPatternsEnd: RegExp[] = sortAttributesEnd.map((sort) => new RegExp(sort));
 
 	const aName = a.name;
 	const bName = b.name;
 
-	let result: CompareResult = 0;
+	const aBeginningIndex: number = sortPatternsBeginning.findIndex((pattern) => pattern.test(aName));
+	const bBeginningIndex: number = sortPatternsBeginning.findIndex((pattern) => pattern.test(bName));
+	const beginning = aBeginningIndex - bBeginningIndex;
 
-	if (result === 0) {
-		let aIndex: number = moveToEnd ? Number.NEGATIVE_INFINITY : Number.POSITIVE_INFINITY;
-		let bIndex: number = moveToEnd ? Number.NEGATIVE_INFINITY : Number.POSITIVE_INFINITY;
-		let aFound = false;
-		let bFound = false;
-		for (let index = 0; index < sortPatterns.length; index++) {
-			const pattern = sortPatterns[index];
-			if (!aFound && pattern.test(aName)) {
-				aIndex = index;
-				aFound = true;
-			}
-			if (!bFound && pattern.test(bName)) {
-				bIndex = index;
-				bFound = true;
-			}
-			if (aFound && bFound) {
-				break;
-			}
-		}
+	if (beginning > 0) return -1;
```

Comment:
Oh I need to improve my eslint...

I really don't like these non curly braces ^^'

Reply:
I'll add those later ;)